### DISCUSSION TITLE
OpenAM 14.6.5 WebAuthn Configuration

### DISCUSSION THREAD

**Discussion by rob3ds** [OpenIdentityPlatform/OpenAM#523] [Q&A] [answered]
Setup OpenAM OOTB default configuration.  Created realm and added / chained WebAuthn Registration and Authentication modules. Set WebAuthn Registration as Required and Authentication chain as requisite.  These settings were based on some very dated online docs...

Problem: Receiving not authenticated error when using registration URI, but have not registered yubikey so I seem to have the cart in front of the horse.

Question: Are there any current / specific guidance on how to properly configure OpenAM 14.x to use WebAuthn modules?

Thanks,
Rob

**Comment by maximthomas** [comment, marked as answer]
Hello, sure https://github.com/OpenIdentityPlatform/OpenAM/wiki/How-to-Setup-WebAuthn-Authentication-in-OpenAM

**Reply by rob3ds** [reply]
Thanks for the quick response.  That is the doc I initially followed.  I indicated (perhaps mistakenly) it seemed dated as the interface is different but settings are all identical.  When I use registration URI: https://host:port/openam/XUI/?org=/realm&service=webauthn-registration I receive an unable to login page with Authentication Error.

**Reply by rob3ds** [reply]
Update: I decided to redo installation and webauthn setup from scratch - environment is 
JRE: jre1.8.0_333
TomEE: tomee-7.1.4-plus 
OpenAM: 14.6.6

After creating new Realm, adding/chaining webauthn modules, now receive "No Configuration Found" when attempting to use registration URI.  

Any directions, suggestions are much appreciated.

Thank you, 
Rob

**Reply by vharseko** [reply]
Try UI vs XUI

**Reply by maximthomas** [reply]
Set system property `-DXUI.enable=false` and restart container to enable classic UI,  just updated the article

**Reply by rob3ds** [reply]
Sorry, hate to be a pita but after adding -DXUI.enable=false I am now receiving a doFilter/forwarding error.  

Partial stack trace:
Exception

javax.servlet.ServletException: AMSetupFilter.doFilter
	com.sun.identity.setup.AMSetupFilter.doFilter(AMSetupFilter.java:141)
	org.forgerock.openam.audit.context.AuditContextFilter.doFilter(AuditContextFilter.java:51)
Root Cause

com.iplanet.jato.NavigationException: Exception encountered during forward
Root cause = [org.owasp.esapi.errors.ConfigurationException: java.lang.reflect.InvocationTargetException Encoder class (org.owasp.esapi.reference.DefaultEncoder) CTOR threw exception.]
	com.iplanet.jato.view.ViewBeanBase.forward(ViewBeanBase.java:380)

Environment:
OS: Windows 2012R2
Java: jre1.8.0_333
TomEE: tomee-7.1.4-plus

**Reply by maximthomas** [reply]
I'll take a look, and try to reproduce the issue. I'll let you know then

**Reply by rob3ds** [reply]
Thank you.  I am sure it is a PICNIC/Cockpit trouble on my side.  I've tried both a default config setup, as well as a custom but using local/embedded opends directory server.

**Reply by maximthomas** [reply]
Have you tried OpenAM docker image?

**Reply by rob3ds** [reply]
I Just downloaded the 14.6.6 zip (OpenAM-14.6.6.zip) from source repo and extracted .war from that. let me try the docker image instead

**Reply by maximthomas** [reply]
Forgot to mention: WebAuthn works only with `https` sites or `localhost` by design

**Reply by rob3ds** [reply]
okay, back on this - I've moved to a new RHEL VM, new setup Tomcat 9, Java 8, Openam14.6.6 - default configuration/embedded setup; still not working by following https://www.openidentityplatform.org/blog/How-to-Setup-WebAuthn-Authentication-in-OpenAM; was curious if the embedded data store required configuration as noted by: https://openam-jp.github.io/WebAuthn/

**Reply by maximthomas** [reply]
Are you still getting the same error? Have you tried docker OpenAM distribution?

**Reply by rob3ds** [reply]
Thanks for the quick response - I will setup docker today. I've attached debug files; I am sure this is cockpit trouble on my side on the config
[Configuration.txt](https://github.com/OpenIdentityPlatform/OpenAM/files/9309786/Configuration.txt)
[Entitlement.txt](https://github.com/OpenIdentityPlatform/OpenAM/files/9309787/Entitlement.txt)
[IdRepo.txt](https://github.com/OpenIdentityPlatform/OpenAM/files/9309788/IdRepo.txt)
[Session.txt](https://github.com/OpenIdentityPlatform/OpenAM/files/9309789/Session.txt)
;

**Reply by rob3ds** [reply]
To answer your original question, a different error: An internal authentication error has occurred.  this is with -DXUI.enabled=false. 

When using https URL I receive error: An internal authentication error has occurred.

When running on localhost using URL:  http://localhost:8080/openam/UI/Login?org=/3dexperience&service=webauthn-registration

I receive this response:

No configuration found.
Contact your system administrator.

[Return to Login page](https://github.com/openam/UI/Login?gx_charset=UTF-8)

This project is licensed under the Common Development and Distribution License (CDDL).
